### ISSUE TITLE
TBG Bug report #1074 (GrantSP on 2015-03-24): Plugin order is being reverted to alphabetical

### ISSUE THREAD

**Issue by TanninOne** [TanninOne/modorganizer#234]
I think there may have been a posting about this that I dismissed as not an issue a few days ago but I can't find that report now.
Apologies to the user that reported it!

I have found that all my MO installations revert the plugin load order to an alphabetical one on running MO each day for the first time.

I can start and use MO numerous times during the course of the day, changing plugin order, adding/removing mods, etc. and see no issues whatsoever.
But each morning when I start my machine and load each MO installation again, the order is alphabetical.
I can't replicate it by changing the date/time settings in Windows and then starting MO, and I haven't yet seen if it applies to system starts through the same day. I'll be doing some tests now after posting this.

It appears to have no bearing on the profile selected. ie. all the profiles, active or not, seem to alphabetised when you choose them.

EDIT: Just tried a couple of system reboots and the order didn't change. Very weird. Only on a new day and only after a system restart but not when you change the date/time during the same day or when you restart on the same day.

# Sort the plugins into a specific working order and close MO
# The next day start MO and see the load order alphabetised.

**Comment by TanninOne** [comment]
GrantSP on 2015-03-24: The issue was updated with the following change(s):
* This issue's description has been changed

**Comment by TanninOne** [comment]
Tannin on 2015-03-24: My guess is that your computer does additional stuff after the first boot of the day like updating your virus scanner / signatures, check for driver updates, and so on.
This is imho the most likely cause it happens only the first try of the day.

Possible reasons are the virus scanner is interfering with MO during its update or the additional load from those background threads causes MO to "stumble".

Can you confirm that that first start of MO takes longer? (including the time the progress bar is displayed at the bottom)
And could you try tomorrow to wait a few minutes after your first boot before starting MO and see if the error still occurs?

**Comment by TanninOne** [comment]
GrantSP on 2015-03-24: 24 hours later and I tried again.
This time only one of the four installations is reverted and it was the second install I started this morning.

All four installs do usually take longer on the first start up, to initialize themselves for the particular game.

As for driver updates or AV updates, nope. I am very careful to only allow those sort of things to happen when I schedule them so that I can check on them.

As I seem to be the only one affected, I left a post on the Nexus asking for people to have a look at this, I think we can safely say this is something to do with my setup.
Let's leave it open and I'll keep a eye on all the matters that I do regarding this phenomenon so we can at least add to the 'weird list'.

**Comment by TanninOne** [comment]
Tannin on 2015-03-25: This is the worst kind of bugs: rare and hard to reproduce but causing serious damage. :(

**Comment by TanninOne** [comment]
GrantSP on 2015-03-26: Well at least its only me and not a wide-spread one.
I didn't notice it straight away because I have been testing small specific test profiles and the order doesn't matter too much if its only the vanilla and maybe 1 or 2 extra.
Then I loaded my actual gaming profiles for each of the games and it then stood out.

No big deal for me, I'll just be sure to check the order before playing instead of launching straight into the games.

**Comment by TanninOne** [comment]
GrantSP on 2015-03-29: Looking into the current profile folder I see '''modlist.txt''' now has the current timestamp but '''loadorder.txt''' also is modified with the same timestamp but it is also of zero length.
Quitting MO immediately changes the '''loadorder.txt''' back to its original size but now it is alphabetized.

Under what conditions should MO be making changes to '''loadorder.txt'''? Only when actual changes are made inside the UI to the load order?

**Comment by TanninOne** [comment]
falcon7370 on 2015-04-01: I am currently having the same exact issue, except I believe mine is worst. Instead of doing this once a day, it does it once every system reboot. It was working fine yesterday, and when I got back on today it started happening. I have no idea why, as I haven't changed anything system-wide between last night and today. What happens, is when MO starts, no plugins are listed. 5 seconds later, it outputs a ton of errors saying "'xxxx'.esp can not be found." for every mod installed, and then it will freeze for about 2-3 seconds, and then they will all appear, disabled, and in alphabetical order. I am going to disable some background processes such as my anti-virus and such to see if it is having any effect but this is making it impossible to set up my game for my new playthrough. :(

**Comment by TanninOne** [comment]
sammael on 2015-04-02: since my report got marked as duplicate I would like to reiterate that I am observing this behaviour NOT only after reboot, but almost consistently 7-8 times out of 10 I run MO regardless of when the last reboot was.

**Comment by TanninOne** [comment]
PaddyGarcia on 2015-04-02: --Edit - I just quit & restarted MO, and the load order remained correct. So it's not 100% of the time.
--Edit #2 - I quit & restarted MO 10 times, (without starting the game) and the load order remained correct. But when I started the game, quit the game, then quit & restarted MO, the load order was scrambled again.

I have a similar problem. This started after I upgraded to MO v.1.3.xx (-AND- after I upgraded to LOOT 0.7? Not sure.)

I know this issue has been reported several times, and it's been "closed, not a bug" several times. Supposedly it's not reproducible, but I reproduce it (nearly) every single time I start MO. (Steps to reproduce: Play FNV, lol.)

Not just once a day, but (nearly) every time I start MO, the plugins are "prioritized" alphabetically. I have to run LOOT (either with MO's integral "sort" button, or with LOOT via MO - both work) to re-sort the plugins every time i start MO. So in my case at least it seemingly has nothing to do with booting the computer. 

(To be clear, I am not making the error of viewing the alphabetized list after clicking the plugins "name" column; with the priorities column selected, plugins that start with "A" have very low priority numbers, as does my Bashed Patch, and my private "ZzTweaks" patch has the highest numerical priority. It really is prioritized alphabetically.) 

After I run LOOT once, I can play the game, quit the game, and restart the game from MO many times, no problem, as long as I keep MO open. But if I close MO, the priorities revert to alphabetical order. 

It's not a crippling bug, except that sometimes I forget to run loot, and that causes subtle problems in-game, as my bashed patch is no longer last etc. (I'm afraid of ending up with a corrupted save at some point.) Actually, I'm surprised the game doesn't crash on start. 

It's not just the way they're listed - the plugins really load in alphabetical order. I know, because many mods don't work correctly until I quit out and sort via LOOT.

**Comment by TanninOne** [comment]
GrantSP on 2015-04-02: Just to clarify something, I haven't closed the other issues because they can't be verified, rather I want all the related experiences recorded here because, regardless of the game or the number of times it occurs, it is the same issue.

I too am seeing it more frequently when starting FNV, perhaps it has something to do with the game specific code implemented in the 1.3.x series?

What makes this frustrating is despite the instances where I see the modlist.txt being of zero-length part way through the process, there doesn't look to be any other logged activity that can be used to pinpoint anything.

Based on recent posts and my further experiences, I think the system start link may be discarded, I don't believe it is related, just coincidental.

**Comment by TanninOne** [comment]
PaddyGarcia on 2015-04-02: --Edit #2 - I just quit & restarted MO 10 times, (without starting the game) and the load order remained correct. But when I started the game, quit the game, then quit & restarted MO, the load order was scrambled again.

--Edit #3 - I just did the above several times, with the same results. Interestingly(?), I only have to load the FNV loading screen (or load a new or saved game, but it's not necessary) to get the bug.

**Comment by TanninOne** [comment]
GrantSP on 2015-04-03: The issue was updated with the following change(s):
* The status has been updated, from '''New''' to '''Confirmed'''.
* This issue's progression has been updated to 10 percent completed.

**Comment by TanninOne** [comment]
GrantSP on 2015-04-03: I can now make this happen at will also. Similar to @PaddyGarcia, whenever I use xEdit and examine the plugins and then quit MO, the next time MO is started the order is changed.
It seems any time the list is examined or worked with the MO sorted list is discarded after quitting from MO.

**Comment by TanninOne** [comment]
Tannin on 2015-04-03: A few questions:
* What version of xedit?
* Do you have MO close when running xedit?
* Could you check the files plugins.txt and loadorder.txt in your profile immediately after running xedit to see if the load order changes right then or on restarting MO?
* Do you have any plugins with "odd" characters in their name like german umlauts, chinese characters, ...?

**Comment by TanninOne** [comment]
GrantSP on 2015-04-03: 
> What version of xedit?
'''3.1.0'''
> Do you have MO close when running xedit?
'''No'''
> Could you check the files plugins.txt and loadorder.txt in your profile
>immediately after running xedit to see if the load order changes right then
>or on restarting MO?
''Plugins.txt'' has a datestamp change, on any function executed inside MO.
''Loadorder.txt'' turns to a zero-length file and ''plugins.txt'' still shows the correct order but the right-hand pane is alphabetised.
Only when MO is closed does the content of ''plugins.txt'' change to alphabetised.
So the change only happens after MO closes, after doing some editing, but files don't appear to be changed until the next time MO starts.
I did notice ''Loadorder.txt'' disappears from the profile folder after closing xEdit.
This file is created as a zero-length file upon starting a new MO session. Immediately after closing this MO session, ''Loadorder.txt'' is filled with plugins in alphabetical order and ''plugins.txt'' is changed at that time to match ''Loadorder.txt''
> Do you have any plugins with "odd" characters in their name like german
>umlauts, chinese characters, ...?
'''No'''

**Comment by TanninOne** [comment]
Tannin on 2015-04-03: Wait, can you pinpoint precisely when the loadorder.txt becomes zero-length?
Also: What game are you testing this with?

To clarify how things in MO work:
* loadorder.txt is the reference for, well, the load order.
* Games before Skyrim did NOT use plugins.txt to specify the order in which esps are loaded, they used the creation time on the file (so the oldest would be loaded first).
* skyrim on the other hand uses the order specified in plugins.txt, which of course only works for enabled plugins, disabled plugins don't have an order in Skyrim.
* Now the devs of wrye bash, NMM and LooT agreed (and I adhered) to use a second file called loadorder.txt which specifies the order of all plugins (disabled and enabled) and synchronize from loadorder.txt to plugins.txt.
* MO uses loadorder.txt for all games and, in case of pre-skyrim games, syncs from loadorder.txt to filetimes.
* Therefore, if the loadorder.txt gets emptied (by whoever) that destroyes the load order. That file is the reference for your load order NOT plugins.txt
* In case of pre-skyrim games, when you run external programs (since they won't support loadorder.txt) MO deletes the loadorder.txt and after running the program regeneratese it by syncing filetimes to loadorder.txt. For Skyrim this shouldn't happen.
* But no matter what, the loadorder.txt should not become zero-length, that is where things actually go wrong.


'''GrantSP wrote:'''
>Plugins.txt has a datestamp change, on any function executed inside MO. 
>Loadorder.txt turns to a zero-length file and plugins.txt still shows the
>correct order but the right-hand pane is alphabetised. 
>Only when MO is closed does the content of plugins.txt change to
>alphabetised. 
>So the change only happens after MO closes, after doing some editing, but
>files don't appear to be changed until the next time MO starts. 
>I did notice Loadorder.txt disappears from the profile folder after closing
>xEdit. 
>This file is created as a zero-length file upon starting a new MO session.
>Immediately after closing this MO session, Loadorder.txt is filled with
>plugins in alphabetical order and plugins.txt is changed at that time to
>match Loadorder.txt

**Comment by TanninOne** [comment]
GrantSP on 2015-04-03: Okay, got it. Very interesting.

Game in question is Fallout 3
# Start MO and if order is wrong run LOOT, or manually reorder, all files look fine and the UI is functioning as normal
# Start xEdit and do whatever in that tool, no edits need actually be made, just using it.
# At that point loadorder.txt disappears, and stays missing even after quitting xEdit and MO.
# Start MO a 2nd time and a zero-length loadorder.txt appears, and obviously the plugins are in the wrong order. Back to point 1 again.

Now that you pointed me in the game-specific direction, I ran some tests on all my installs and you are correct, it is only the non-Skyrim games that this happens. I can make it happen like clockwork for TES4 & Fallout3/NV, but it doesn't seem to affect TES5.

Either xEdit or another tool has changed the datestamp on all my plugins for these games and they are all dated sometime with a year 2000. With, as you'd expect, the dates getting larger as the name of the plugin is further into the alphabet.
I haven't used any of the Wrye Bash tools lately on these games, though I have at one time done so whilst testing its functionality. That tool I believe modifies the dates and it may just have become evident with the changes in 1.3.x?

Curiously though, I did use WB more for Skyrim and there is no change to the dates of these plugins. Possibly due to the more extensive work being done with that code than the other variants.
If I had been a little more observant when first noticing this, we may have had this licked by now! Examine things more closely should be my motto, not speak first then look. :)

Any further tests you want, just whistle.

**Comment by TanninOne** [comment]
GrantSP on 2015-04-13: Version 1.3.4 appears to solved this issue.

**Comment by TanninOne** [comment]
simsrw73 on 2015-05-08: This has been happening to me also in the 1.3.x series (currently 1.3.4) for Skyrim. It happened again today which is really a problem for me because I sort my mods by hand. I don't know if it's best or not, but I don's use LOOT or any sorter.

I hadn't played Skyrim in a few weeks. I started it up a couple days ago. I upgraded a few mods, installed a few new ones and manually sorted the new ones in to place. I also upgraded TES5Edit. MO & TES5Edit are stored in the Skyrim directory. I also used MO to delete a couple of old profiles that were copies of my main profile. All of the previous happened over the last couple of days. I don't believe I closed MO during any of that (but I don't trust my memory much!). I know that I had run and tested my final changes and everything tested fine. Just now I switched over to MO. It showed that there was a mod conflict which wasn't indicated before. I clicked on the List Options icon and selected Refresh because there was no other indication of the supposed conflict.  Now nearly all of my plugins are unselected and simply sorted alphabetically. loadorder.txt is resorted. plugins.txt is truncated.

I'm not positive on the sequence of events or if they were the same each time, but this has happened a number of times since upgrading to the 1.3.x series. Fortunately, most of the time I had a recent backup. Unfortunately, this time I don't and I'm seriously worried about getting my game back in order. I did immediately make a manual backup of my profile directory as soon as I realized this happened, but it is tainted also. Is there any kind of logging done by MO to help track this down? MO did make a backup copy of loadorder.txt/plugins.txt that is unfortunately a bit over a month old.

I will be going out of town for the weekend, but I will be happy to do anything I can to help track this problem down beginning next week. I will definitely work on finding a reproducible case. This is definitely a current and active bug.

[[https://dl.dropboxusercontent.com/u/5987573/logs.zip|Logs]] // 
[[https://dl.dropboxusercontent.com/u/5987573/Torandis%20%28Current%29.zip|Profile]]

**Comment by ThosRTanner** [comment]
Request - could you change the # and 2 to something else in one of your earlier comments? github thinks you're referring to a pull request(!)
